contracts/RMRK/equippable/RMRKEquippable.sol

```diff
@@ -373,16 +375,17 @@ contract RMRKEquippable is RMRKNesting, IRMRKEquippable {
         internal
         virtual
     {
-        _tokenApprovalsForResources[tokenId] = to;
-        emit ApprovalForResources(ownerOf(tokenId), to, tokenId);
+        address owner = ownerOf(tokenId);
+        _tokenApprovalsForResources[tokenId][owner] = to;
+        emit ApprovalForResources(owner, to, tokenId);
     }
 
-    function _cleanApprovals(address owner, uint256 tokenId)
+    function _cleanApprovals(uint256 tokenId)
```

Comment:
We could also clear all of the approvals on the token using this method, not only the approvals set by current owner 🤔

Reply:
Clearing all approvals might be safer for the user as then we want have the original old permissions back.

> // WARNING: If a child NFT returns the original root owner, old permissions would be active again

Reply:
That would be ideal, but you cannot clear a mapping. :smiling_face_with_tear:

Reply:
One thing we could do to improve this warning situation is to also clear approvals on incoming NFTs. It will change nothing for nested NFTs though. It would just help if I approved something, nested into another NFT, it left my wallet and it came back unnested. :/